src/util/include/util/Color.h

```diff
@@ -11,149 +11,160 @@
 
 #pragma once
 
+#include <algorithm>
 #include <cassert>
 #include <cstdint>
-#include <limits>
-
-#include <gtk/gtk.h>
-
-#ifndef XOURNAL_ENFORCE_COLOR
-using Color = uint32_t;
-#else
+#include <cstring>
 #include <iostream>
 #include <limits>
 #include <type_traits>
-struct Color {
-    constexpr Color() = default;
-
-    template <typename T, std::enable_if_t<std::is_same_v<T, uint32_t>, int> = 0>
-    constexpr Color(T t): val(uint32_t(t)) {}
-    template <typename T, std::enable_if_t<std::is_unsigned_v<T> && !std::is_same_v<T, uint32_t>, int> = 0>
-    constexpr explicit Color(T t): val(uint32_t(t)) {}
-    template <typename T, std::enable_if_t<std::is_signed_v<T>, int> = 0>
-    constexpr explicit Color(T t): val(uint32_t(t)) {}
 
-    constexpr explicit operator uint32_t&() { return val; }
+#include <gtk/gtk.h>
+#include <stdint.h>
+
+struct LegacyARGBU8Color {
+    constexpr LegacyARGBU8Color(): val(){};  // NOLINT(cppcoreguidelines-pro-type-member-init)
+    constexpr LegacyARGBU8Color(uint8_t r, uint8_t g, uint8_t b, uint8_t a = 0): b(b), g(g), r(r), a(a) {}
```

Comment:
The thing is, the order does not change the initialisation order. Values are initialized in the same order as they appear in the Fields. Since this is the Legacy Color struct, and it must be the same as 0x[AA][RR][GG][BB], it would change nothing at all. Also, a dev could think, r is initialized before b, but actually it's the other way round.

Reply:
With the endianness argument below it makes sense. Thanks.